### PR TITLE
feat: user search page and add nuance to friendships

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR
Reorganized user and friend management functionality, adding user search capabilities and friend status tracking.

### What changed?
- Renamed `friends` directory to `users` to better reflect its broader functionality
- Added `isFriend` boolean flag to user objects
- Created new `useUserSearch` hook for searching users
- Replaced `useFriend` with more generic `useUser` hook
- Implemented user search functionality in the add-friends screen
- Updated Profile component to use `isFriend` status from user data
- Renamed `useAddFriend` to `useSendFriendRequest` to better reflect its purpose

### How to test?
1. Navigate to the Friends tab
2. Try adding new friends using the search functionality
3. Verify friend status is correctly displayed for each user
4. Test friend management actions (add/remove friend)
5. Confirm user search returns appropriate results based on username or display name
6. Verify friend request messages display correctly when adding friends

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/136?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#136** <a href="https://app.graphite.dev/github/pr/habits-together/app/136?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/136?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#135** <a href="https://app.graphite.dev/github/pr/habits-together/app/135?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#134** <a href="https://app.graphite.dev/github/pr/habits-together/app/134?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#133** <a href="https://app.graphite.dev/github/pr/habits-together/app/133?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#132** <a href="https://app.graphite.dev/github/pr/habits-together/app/132?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — APPROVED** [review]
